bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuntimeRule.java

```diff
@@ -28,6 +33,7 @@
 public class RuntimeRule extends Rule {
 
     private Map<String, Module> moduleMap;
+    private RuleStatusInfo statusInfo;
```

Comment:
Actually each runtime rule must always have a status. I think it should be part of it. In these case there is no need to synchronize maps of runtime rules and their statuses.